File: src/main/java/org/gangel/graphomance/neo4j/NeoConnection.java
package org.gangel.graphomance.neo4j;

import lombok.Builder;
import lombok.Getter;
import org.gangel.graphomance.Connection;
import org.neo4j.graphdb.GraphDatabaseService;

@Builder
public class NeoConnection implements Connection {

    @Getter
    private GraphDatabaseService dbService;

    @Override
    public void open() {

    }

    @Override
    public void close() {
        if (dbService != null) {
            dbService.shutdown();
            dbService = null;
        }
    }
}

File: src/main/kotlin/org/gangel/graphomance/neo4j/NeoConnection.kt
package org.gangel.graphomance.neo4j

import org.gangel.graphomance.Connection
import org.neo4j.driver.Driver
import org.neo4j.driver.Session

class NeoConnection(private val driver: Driver) : Connection {
	override fun open() {}
	override fun close() {}

	fun session() : Session = driver.session()
}
